python/tskit/tables.py

```diff
@@ -582,6 +582,20 @@ def append(self, row):
             }
         )
 
+    def replace_with(self, other):
+        """
+        Entirely replace the contents of this table with another
```

Comment:
Can you make it clear here that the contents of this table are overwritten by copying the other table?
Then turn it into a comment as I'd like to keep this undocumented for now so that we can consider if proper assignment is the right solution.